---
int32 id            #id of taken scan
string message
bool success